lib/src/cpp/utils/ParseXML.cpp

```diff
@@ -399,18 +431,37 @@ std::string ParseXML::get_absolute_path(
     return absolute_xml_file;
 }
 
+std::vector<uint>* ParseXML::get_real_index(
+        xercesc::DOMNodeList*& node_list)
+{
+    // Create new list
+    std::vector<uint>* index_list = new std::vector<uint>();
```

Comment:
This is really dangerous. You are allocating memory in the heap which is going to be freed at scope exit (after the return command). The return gives the pointer to the allocated memory that now has been freed. The program can reuse this memory at any time, invalidating the information pointed by the pointer. In this case you should use smart pointers (`std::unique_ptr`).